cannot run j2cl:watch on win10 21h2

C:\Code\jdk-18+36\bin\java.exe -Dmaven.multiModuleProjectDirectory=D:\Code\Java\Projects\java2Js\compiler "-Dmaven.home=C:\Code\Ide\IntelliJ IDEA 2022.1\plugins\maven\lib\maven3" "-Dclassworlds.conf=C:\Code\Ide\IntelliJ IDEA 2022.1\plugins\maven\lib\maven3\bin\m2.conf" "-Dmaven.ext.class.path=C:\Code\Ide\IntelliJ IDEA 2022.1\plugins\maven\lib\maven-event-listener.jar" "-javaagent:C:\Code\Ide\IntelliJ IDEA 2022.1\lib\idea_rt.jar=52818:C:\Code\Ide\IntelliJ IDEA 2022.1\bin" -Dfile.encoding=UTF-8 -classpath "C:\Code\Ide\IntelliJ IDEA 2022.1\plugins\maven\lib\maven3\boot\plexus-classworlds-2.6.0.jar;C:\Code\Ide\IntelliJ IDEA 2022.1\plugins\maven\lib\maven3\boot\plexus-classworlds.license" org.codehaus.classworlds.Launcher -Didea.version=2022.1 com.vertispan.j2cl:j2cl-maven-plugin:0.19:watch
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------------< org.example:compiler >------------------------
[INFO] Building compiler 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- j2cl-maven-plugin:0.19:watch (default-cli) @ compiler ---
[INFO] Using D:\Code\Java\Projects\java2Js\compiler\target/compiler-1.0-SNAPSHOT as webappDirectory since goal is running in a non-reactor build and none was set
[INFO] Removing dependency com.google.jsinterop:base:jar:1.0.0:compile, replacing with com.vertispan.jsinterop:base:jar:1.0.0-1
[INFO] Removing dependency com.google.jsinterop:jsinterop-annotations:jar:1.0.2:compile, no replacement
[INFO] Starting com.vertispan.jsinterop:base:1.0.0-1/unpack
[INFO] Finished com.vertispan.jsinterop:base:1.0.0-1/unpack in 6ms
[INFO] Start watching {org.example:compiler:1.0-SNAPSHOT=[D:\Code\Java\Projects\java2Js\compiler\src\main\java, D:\Code\Java\Projects\java2Js\compiler\src\main\resources]}
[INFO] Watching for changes
[INFO] Starting com.vertispan.jsinterop:base:1.0.0-1/bytecode
[INFO] Finished com.vertispan.jsinterop:base:1.0.0-1/bytecode in 9ms
[INFO] Starting org.example:compiler:1.0-SNAPSHOT/bytecode
[INFO] Starting com.vertispan.jsinterop:base:1.0.0-1/stripped_sources
[INFO] Finished com.vertispan.jsinterop:base:1.0.0-1/stripped_sources in 61ms
[INFO] Starting com.vertispan.jsinterop:base:1.0.0-1/stripped_bytecode
[INFO] Starting com.vertispan.jsinterop:base:1.0.0-1/transpiled_js
[INFO] Finished org.example:compiler:1.0-SNAPSHOT/bytecode in 527ms
[INFO] Starting org.example:compiler:1.0-SNAPSHOT/stripped_sources
[INFO] Finished com.vertispan.jsinterop:base:1.0.0-1/stripped_bytecode in 469ms
[INFO] Starting com.vertispan.jsinterop:base:1.0.0-1/stripped_bytecode_headers
[INFO] Starting org.example:compiler:1.0-SNAPSHOT/transpiled_js
[INFO] Finished org.example:compiler:1.0-SNAPSHOT/transpiled_js in 249ms
[INFO] Finished com.vertispan.jsinterop:base:1.0.0-1/transpiled_js in 754ms
[INFO] Starting org.example:compiler:1.0-SNAPSHOT/optimized_js
java.lang.IllegalArgumentException: 'other' has different root
	at java.base/sun.nio.fs.WindowsPath.relativize(WindowsPath.java:420)
	at java.base/sun.nio.fs.WindowsPath.relativize(WindowsPath.java:42)
	at com.google.javascript.jscomp.CommandLineRunner.prepForBundleAndAppendTo(CommandLineRunner.java:2046)
	at com.google.javascript.jscomp.AbstractCommandLineRunner.printBundleTo(AbstractCommandLineRunner.java:2339)
	at com.google.javascript.jscomp.AbstractCommandLineRunner.outputManifestOrBundle(AbstractCommandLineRunner.java:2203)
	at com.google.javascript.jscomp.AbstractCommandLineRunner.outputBundle(AbstractCommandLineRunner.java:2153)
	at com.google.javascript.jscomp.AbstractCommandLineRunner.processResults(AbstractCommandLineRunner.java:1565)
	at com.google.javascript.jscomp.AbstractCommandLineRunner.doRun(AbstractCommandLineRunner.java:1324)
	at com.google.javascript.jscomp.AbstractCommandLineRunner.run(AbstractCommandLineRunner.java:535)
	at com.vertispan.j2cl.tools.Closure.compile(Closure.java:183)
	at com.vertispan.j2cl.build.provided.ClosureTask$4.execute(ClosureTask.java:225)
	at com.vertispan.j2cl.build.TaskScheduler$2.executeTask(TaskScheduler.java:172)
	at com.vertispan.j2cl.build.TaskScheduler$2.lambda$onReady$0(TaskScheduler.java:211)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
java.lang.RuntimeException
	at com.vertispan.j2cl.build.DiskCache.markFailed(DiskCache.java:452)
	at com.vertispan.j2cl.build.DiskCache$CacheResult.markFailure(DiskCache.java:54)
	at com.vertispan.j2cl.build.TaskScheduler$2.executeTask(TaskScheduler.java:181)
	at com.vertispan.j2cl.build.TaskScheduler$2.lambda$onReady$0(TaskScheduler.java:211)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
[ERROR] Exception executing task org.example:compiler:1.0-SNAPSHOT/optimized_js
java.lang.IllegalStateException: Closure Compiler failed, check log for details
    at com.vertispan.j2cl.build.provided.ClosureTask$4.execute (ClosureTask.java:245)
    at com.vertispan.j2cl.build.TaskScheduler$2.executeTask (TaskScheduler.java:172)
    at com.vertispan.j2cl.build.TaskScheduler$2.lambda$onReady$0 (TaskScheduler.java:211)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:539)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run (ScheduledThreadPoolExecutor.java:304)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:833)

--- a/j2cl-tasks/src/main/java/com/vertispan/j2cl/build/provided/ClosureBundleTask.java
+++ b/j2cl-tasks/src/main/java/com/vertispan/j2cl/build/provided/ClosureBundleTask.java
@@ -98,7 +98,6 @@ public class ClosureBundleTask extends TaskFactory {
                     Collections.emptyMap(),
                     Collections.emptyList(),//TODO actually pass these in when we can restrict and cache them sanely
                     Optional.empty(),
-                    null,
                     true,//TODO have this be passed in,
                     true,//default to true, will have no effect anyway
                     false,

--- a/j2cl-tasks/src/main/java/com/vertispan/j2cl/build/provided/ClosureTask.java
+++ b/j2cl-tasks/src/main/java/com/vertispan/j2cl/build/provided/ClosureTask.java
@@ -234,7 +234,6 @@ public class ClosureTask extends TaskFactory {
                         defines,
                         externs,
                         translationsfile,
-                        null,
                         true,//TODO have this be passed in,
                         checkAssertions,
                         rewritePolyfills,

--- a/j2cl-tasks/src/main/java/com/vertispan/j2cl/build/provided/JsZipBundleTask.java
+++ b/j2cl-tasks/src/main/java/com/vertispan/j2cl/build/provided/JsZipBundleTask.java
@@ -56,7 +56,6 @@ public class JsZipBundleTask extends TaskFactory {
                     Collections.emptyMap(),
                     Collections.emptyList(),
                     Optional.empty(),
-                    null,
                     true,
                     true,
                     false,

--- a/j2cl-tasks/src/main/java/com/vertispan/j2cl/tools/Closure.java
+++ b/j2cl-tasks/src/main/java/com/vertispan/j2cl/tools/Closure.java
@@ -55,7 +55,6 @@ public class Closure {
             Map<String, String> defines,
             Collection<String> externFiles,
             Optional<File> translationsFile,
-            PersistentInputStore persistentInputStore,
             boolean exportTestFunctions,
             boolean checkAssertions,
             boolean rewritePolyfills,
@@ -66,7 +65,6 @@ public class Closure {
         List<String> jscompArgs = new ArrayList<>();
 
         Compiler jsCompiler = new Compiler(System.err);
-//        jsCompiler.setPersistentInputStore(persistentInputStore);
 
         // List the parent directories of each input so that module resolution works as expected
         jsInputs.keySet().forEach(parentPath -> {
@@ -187,21 +185,12 @@ public class Closure {
             return false;
         }
 
-        //TODO historically we didnt populate the persistent input store until this point, so put it here
-        //     if we restore it
+        jscompRunner.run();
 
-        try {
-            jscompRunner.run();
-
-            if (jscompRunner.hasErrors() || jscompRunner.exitCode != 0) {
-                return false;
-            }
-        } finally {
-            if (jsCompiler.getModules() != null) {
-                // clear out the compiler input for the next go-around
-                jsCompiler.resetCompilerInput();
-            }
+        if (jscompRunner.hasErrors() || jscompRunner.exitCode != 0) {
+            return false;
         }
+
         return true;
     }
 

--- a/pom.xml
+++ b/pom.xml
@@ -88,7 +88,7 @@
 
     <!-- Builder dependencies versions -->
     <j2cl.version>0.10.0-3c97afeac</j2cl.version>
-    <closure.compiler.unshaded.version>v20210808-1</closure.compiler.unshaded.version>
+    <closure.compiler.unshaded.version>v20220502-1</closure.compiler.unshaded.version>
     <commons.codec.version>1.11</commons.codec.version>
     <commons.io.version>2.7</commons.io.version>
 
@@ -101,7 +101,7 @@
     <plexus.utils.version>3.1.0</plexus.utils.version>
     <plexus.classworlds.version>2.5.2</plexus.classworlds.version>
     <plexus.interpolation.version>1.25</plexus.interpolation.version>
-    <guava.version>30.0-jre</guava.version>
+    <guava.version>31.1-jre</guava.version>
     <error.prone.annotations.version>2.1.3</error.prone.annotations.version>
     <jsr305.version>3.0.2</jsr305.version>
     <commons.lang3.version>3.8.1</commons.lang3.version>
@@ -194,29 +194,30 @@
       </dependency>
 
       <dependency>
-        <groupId>com.vertispan.javascript</groupId>
-        <artifactId>closure-compiler-unshaded</artifactId>
-        <version>${closure.compiler.unshaded.version}</version>
+        <groupId>com.vertispan.j2cl</groupId>
+        <artifactId>backend-libraryinfo</artifactId>
+        <version>${j2cl.version}</version>
         <exclusions>
-          <exclusion>
-            <groupId>com.google.jsinterop</groupId>
-            <artifactId>jsinterop-annotations</artifactId>
-          </exclusion>
           <exclusion>
             <groupId>com.google.protobuf</groupId>
             <artifactId>protobuf-java</artifactId>
           </exclusion>
           <exclusion>
-            <groupId>args4j</groupId>
-            <artifactId>args4j</artifactId>
+            <groupId>com.google.protobuf</groupId>
+            <artifactId>protobuf-java-util</artifactId>
           </exclusion>
+        </exclusions>
+      </dependency>
+
+
+      <dependency>
+        <groupId>com.vertispan.javascript</groupId>
+        <artifactId>closure-compiler-unshaded</artifactId>
+        <version>${closure.compiler.unshaded.version}</version>
+        <exclusions>
           <exclusion>
-            <groupId>com.google.jsinterop</groupId>
-            <artifactId>base</artifactId>
-          </exclusion>
-          <exclusion>
-            <groupId>com.google.elemental2</groupId>
-            <artifactId>*</artifactId>
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
           </exclusion>
         </exclusions>
       </dependency>

